src/App/Fossa/Analyze/Debug.hs

```diff
@@ -54,8 +56,51 @@ import GHC.Conc qualified as Conc
 import GHC.Generics (Generic)
 import GHC.Stats qualified as Stats
 import System.Args (getCommandArgs)
+import System.Environment (getEnvironment)
 import System.Info qualified as Info
 
+collectEnvVariables :: Has (Lift IO) sig m => m (Map.Map Text Text)
+collectEnvVariables =
+  (Map.filterWithKey onlyEnvOfInterest <$> Map.fromList)
+    . map (bimap toText toText)
+    <$> sendIO getEnvironment
+  where
+    -- environment variables are case-sensitive in linux,
+    -- but are not case sensitive in windows, so always
+    -- perform case-insensitive comparison
+    onlyEnvOfInterest :: Text -> Text -> Bool
```

Comment:
I decided against including all environment variables by default in case credentials or authorization keys are included - e.g. `AWS_ACCESS_KEY_ID` etc.

Reply:
Agreed, if any of these GO env vars have the potential to include any type of secrets we should exclude them as well.